docs/src/modules/components/Demo.js

```diff
@@ -173,28 +80,191 @@ function useUniqueId(prefix) {
 }
 
-function Demo(props) {
-  const { classes, demo, demoOptions, githubLocation } = props;
-  const dispatch = useDispatch();
-  const t = useSelector((state) => state.options.t);
-  const codeVariant = useSelector((state) => state.options.codeVariant);
-  const demoData = getDemoData(codeVariant, demo, githubLocation);
-
-  const [sourceHintSeen, setSourceHintSeen] = React.useState(false);
+const useDemoToolbarStyles = makeStyles(
+  (theme) => {
+    return {
+      root: {
+        display: 'none',
+        [theme.breakpoints.up('sm')]: {
+          display: 'flex',
+          flip: false,
+          top: 0,
+          right: theme.spacing(1),
+          height: theme.spacing(6),
+        },
+        justifyContent: 'space-between',
+      },
+      toggleButtonGroup: {
+        margin: '8px 0',
+      },
+      toggleButton: {
+        height: 32,
+      },
+      tooltip: {
+        zIndex: theme.zIndex.appBar - 1,
+      },
+    };
+  },
+  { name: 'DemoToolbar' },
+);
+
+const alwaysTrue = () => true;
+
+/**
+ *
+ * @param {React.Ref<HTMLElement>[]} controlRefs
+ * @param {object} [options]
+ * @param {(index: number) => boolean} [options.isFocusableControl] In case certain controls become unfocusable
+ * @param {number} [options.defaultActiveIndex]
+ */
+function useToolbar(controlRefs, options = {}) {
+  const { defaultActiveIndex = 0, isFocusableControl = alwaysTrue } = options;
+  const [activeControlIndex, setActiveControlIndex] = React.useState(defaultActiveIndex);
+
+  // TODO: do we need to do this during layout practically? It's technically
+  // a bit too late since we allow user interaction between layout and passive effects
   React.useEffect(() => {
-    setSourceHintSeen(getCookie('sourceHintSeen'));
-  }, []);
+    setActiveControlIndex((currentActiveControlIndex) => {
+      if (!isFocusableControl(currentActiveControlIndex)) {
+        return defaultActiveIndex;
+      }
+      return currentActiveControlIndex;
+    });
+  }, [defaultActiveIndex, isFocusableControl]);
+
+  // controlRefs.findIndex(controlRef => controlRef.current = element)
+  function findControlIndex(element) {
+    let controlIndex = -1;
+    controlRefs.forEach((controlRef, index) => {
+      if (controlRef.current === element) {
+        controlIndex = index;
+      }
+    });
+    return controlIndex;
+  }
 
-  const [demoHovered, setDemoHovered] = React.useState(false);
-  const handleDemoHover = (event) => {
-    setDemoHovered(event.type === 'mouseenter');
+  function handleControlFocus(event) {
+    const nextActiveControlIndex = findControlIndex(event.target);
+    if (nextActiveControlIndex !== -1) {
+      setActiveControlIndex(nextActiveControlIndex);
+    } else {
+      // make sure DCE works
+      // eslint-disable-next-line no-lonely-if
+      if (process.env.NODE_ENV !== 'production') {
+        console.error(
+          'Material-UI: The toolbar contains a focusable element that is not controlled by the toolbar. ' +
```

Comment:
```suggestion
          'Material-UI: the toolbar contains a focusable element that is not controlled by the toolbar. ' +
```

Reply:
Why lowercase at the beginning of a sentenceß

Reply:
The motivation for the comment was to match the other components. But considering: 1. it's the beginning of a sentence 2. React does handle it identically: https://github.com/facebook/react/search?p=4&q=toErrorDev%28&unscoped_q=toErrorDev%28. It sounds like we should migrate the codebase :)